bids/layout/index.py

```diff
@@ -26,16 +26,22 @@ def _extract_entities(bidsfile, entities):
     return match_vals
 
 
-def _check_path_matches_patterns(path, patterns):
+def _check_path_matches_patterns(path, patterns, root=None):
     """Check if the path matches at least one of the provided patterns. """
     if not patterns:
         return False
+
     path = path.absolute()
+    if root is not None:
+        path = Path("/") / path.relative_to(root)
+
+    # Path now can be downcast to str
+    path = str(path)
     for patt in patterns:
-        if isinstance(patt, Path):
-            if path == patt:
+        if isinstance(patt, (str, Path)):
+            if str(patt) in path:
```

Comment:
@effigies I hope I did not misinterpreted these two lines.

Here is the string matching pattern:
1) Shouldn't we allow strings, not just Paths?
2) Shouldn't we just check the presence of the substring, rather than the exact match?

Reply:
Finally, the changes were much more involved - probably this doesn't have much relevance now.